libunwind/src/UnwindRegistersRestore.S

```diff
@@ -703,6 +703,25 @@ Lnovec:
 
 #elif defined(__aarch64__)
 
+#if defined(__CHERI_PURE_CAPABILITY__)
```

Comment:
Should this not be gated by _LIBUNWIND_SANDBOX_OTYPES?

Reply:
It works fine as-is on a CheriBSD rtld that doesn't support c18n, but if the goal is to ifdef the whole thing out and keep libunwind behaving exactly the way it did before (in the sense that not defining `_LIBUNWIND_SANDBOX_OTYPES` causes _all_ the changes to not be compiled in and we're left with the source of libunwind pre-c18n support 1:1) we could put it behind an `#ifdef _LIBUNWIND_SANDBOX_OTYPES` as it simply won't be used unless that is defined.

Reply:
I mean, part of the point is that __rtld_foo isn't going to be the naming convention for other OS's dynamic linkers.

Reply:
ACK. I'm leaning towards leaving it as-is for now because this will likely change as we redesign the APIs post-release anyway. This bit of the code needs to have a cleaner ABI that doesn't depend on the structure of the unwind context. It's very much tied to both libunwind and CheriBSD's rtld. We just need to make sure to document this in the c18n man page.